tools/nightly_msrv.sh

```diff
@@ -0,0 +1,101 @@
+#! /bin/bash
+
+#
+# Ensure all necessary dependencies are installed
+#
+
+if ! command -v rustup >/dev/null 2>&1; then
```

Comment:
nit: will we really meet this situation? IIRC our CI has a step of installing rustup etc. and for local users, if someone does not have rustup, how can he/she develop frb ;)

Reply:
You are correct in your thinking, HOWEVER, remember this is bash. If a command isn't found, the script continues anyway, potentially causing rather large issues.

three lines can make CERTAIN rustup is available on the system with a clear mind.

Perhaps someone is using a new shell that does not have `.rustup/` in the shell-loaded path?

It is an edge case certainly, but it is common good practice to ensure all tools are installed at the beginning of bash scripts for this reason.

Reply:
> HOWEVER, remember this is bash. If a command isn't found, the script continues anyway, potentially causing rather large issues.

I have learnt to set `set -euxo pipefail` *whenever* you are writing any bash script. Then if a command is not found (or bad return code, or whatever), you are guaranteed to fail immediately.